bundle/regal/rules/bugs/constant-condition/constant_condition.rego

```diff
@@ -9,36 +9,31 @@ import data.regal.result
 # Additionally, there are several other conditions that could be considered
 # constant, or if not, redundant... so this rule should be expanded in time
 
-_operators := {"equal", "gt", "gte", "lt", "lte", "neq"}
-
-_rules_with_bodies := [rule |
-	some rule in input.rules
-	rule.body
-]
-
 # METADATA
 # description: single scalar value, like a lone `true` inside a rule body
 # scope: rule
 report contains violation if {
-	expr := _rules_with_bodies[_].body[_]
+	terms := input.rules[_].body[_].terms
 
 	# We could probably include arrays and objects too, as a single compound value
 	# is not very useful, but it's not as clear cut as scalars, as you could have
 	# something like {"a": foo(input.x) == "bar"} which is not a constant condition,
 	# however meaningless it may be. Maybe consider for another rule?
-	expr.terms.type in ast.scalar_types
+	terms.type in ast.scalar_types
 
-	violation := result.fail(rego.metadata.chain(), result.location(expr.terms))
+	violation := result.fail(rego.metadata.chain(), result.location(terms))
 }
 
 # METADATA
 # description: two scalar values with a "boolean operator" between, like 1 == 1, or 2 > 1
 # scope: rule
 report contains violation if {
-	expr := _rules_with_bodies[_].body[_]
+	operators := {"equal", "gt", "gte", "lt", "lte", "neq"}
+
+	expr := input.rules[_].body[_]
```

Comment:
Hm! I'd have expected the linter to scream "use 'some .. in ..' here"

Reply:
There's an exception made for "nested" iteration. It can even be configured for those that want it 🙂